docs/sections/workflows/workchain.md

```diff
@@ -14,123 +14,161 @@ In this module, you will learn step-by-step how to write work chains in AiiDA.
+
+* The `x` input is obtained from the inputs using `self.inputs.x`, i.e. as an _attribute_ from `self.inputs`.
+* Using the `out()` method, this input is attached as an output of the work chain with link label `workchain_result`.
+  As the `x` work chain input is already an `Int` node, the `valid_type` condition is immediately satisfied.
 
-Now we should be ready to run our first work chain!
+Add the `result` method to the `OutputInputWorkChain` class in your `my_first_workchain.py` file.
+Now you should be ready to run your first work chain!
 
 ### Run the work chain
 
-In the terminal, navigate to the folder where you saved the script with your first work chain and open a `verdi shell`:
+In the terminal, navigate to the folder where you saved the `my_first_workchain.py` file with the `OutputInputWorkChain` work chain and open a `verdi shell`.
+Then run the work chain as you have seen in the **TODO ADD LINK**:
 
 ```{code-block} ipython
 In [1]: from aiida.engine import run
 In [2]: from my_first_workchain import OutputInputWorkChain
 In [3]: result = run(OutputInputWorkChain, x=Int(4) )
-In [4]: result
-Out[2]:
-{'product': <Int: uuid: 9c5c29de-6176-41fe-a051-672a5348e631 (pk: 1909) value: 4>}
 ```
 
-Note that, if you had tried to pass as input an integer, for example with:
+This should complete almost instantaneously, we're just passing an input after all!
+Let's see what's stored in the `result` variable:
+
 ```{code-block} ipython
-...
-In [3]: result = run(OutputInputWorkChain, x=4 )
+In [4]: result
+Out[4]:
+{'workchain_result': <Int: uuid: ed5106ef-8eff-4e87-b2e9-ce6770a6b9a3 (pk: 4665) value: 4>}
 ```
-that would have raised an error as a Python integer is not an AiiDA data type.
 
-Exit the `verdi shell` and check the list of processes with `verdi process list`, using `-a/--all` to also see _terminated_ processes and `-p/--past-days 1` to only see processes _created_ in the past day:
+You can see that the `run()` method has returned a dictionary whose _key_ corresponds to the output link label (`'workchain_result'`) and whose _value_ is the `Int` node that has been passed as an output in the work chain.
+
+Exit the `verdi shell` and check the list of processes with `verdi process list`, using `-a/--all` to also see all _terminated_ processes and `-p/--past-days 1` to only see processes _created_ in the past day:
 ```{code-block} console
 $ verdi process list -a -p 1
 ...
 1982  2m ago     OutputInputWorkChain  ⏹ Finished [0]
 ```
-Also check the status of the process that corresponds to the work chain `OutputInputWorkChain`:
-```{code-block} console
-$ verdi process process status 1982
-OutputInputWorkChain<1982> Finished [0] [None]
-```
-And show some details about the inputs and outputs with:
+
+Grab the PK of the `OutputInputWorkChain` and show some details about the inputs and outputs using `verdi process show`:
+
 ```{code-block} console
-$ verdi process show 1982
+$ verdi process show <PK>
```

Comment:
Do you know why when we use the snippet button to copy this command, we get the command plus two blank lines? This makes the command to be executed automatically when pasted on a terminal.

Reply:
Hmm, Indeed. Let me check the `sphinx-copybutton` regex again.

Reply:
Strange, if I test this via https://regex101.com/, it seems to only give 1 match, so not sure where the added empty lines come from.

I suggest we simply fix this by splitting up the in- and output in two code-blocks for now.

Reply:
Alright, the empty lines are added as well to the copied content. Might need to raise an issue on https://github.com/executablebooks/sphinx-copybutton, but for now we just keep this in mind for code snippets.

Reply:
Alright then.